File: storeCashFlowServices/store-cash-flow-api/src/main/java/com/idApps/storeCashFlowApi/models/response/ApiResponse.java
package com.idApps.storeCashFlowApi.models.response;

import lombok.Data;

@Data
public class ApiResponse<T> {
    private boolean success;
    private T data;
    private String error;

    public ApiResponse(T data) {
        this.success = true;
        this.data = data;
    }
    public ApiResponse(Exception error) {
        this.success = false;
        this.error = error.getMessage();
    }
}

File: store_cash_flow_front/src/app/models/api-response.ts
export class ApiResponse<T> {
    success: boolean;
    data?: T;
    error?: string;

    constructor(dataOrError: T | Error) {
        if (dataOrError instanceof Error) {
            this.success = false;
            this.error = dataOrError.message;
        } else {
            this.success = true;
            this.data = dataOrError;
        }
    }
}